packages/task/src/browser/task-schema-updater.ts

```diff
@@ -70,12 +72,15 @@ export class TaskSchemaUpdater {
         this.taskDefinitionRegistry.onDidUnregisterTaskDefinition(() => this.updateSupportedTaskTypes());
     }
 
-    update(): void {
+    readonly update = debounce(() => this.doUpdate(), 0);
```

Comment:
use `setImmediate` instead to make sure that events are fired at the end of the same tick, not in next tick

Reply:
not really matter here